sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -72,6 +72,18 @@ abstract class UnaryMathExpression(f: Double => Double, name: String)
   }
 }
 
+// for floor and ceil which returns bigint instead of double
+abstract class UnaryMathExpressionWithBigIntRet(f: Double => Double, name: String)
```

Comment:
Did you consider parameterizing `UnaryMathExpression` by the return type as opposed to using inheritance?